### ISSUE TITLE
VSCode  with Python plugin automatically executing a carriage return

### ISSUE THREAD

**Issue by bruparel** [DonJayamanne/pythonVSCode#804]
## Environment data
VS Code version:  1.10.2
Python Extension version:  0.5.9
Python Version:   2.7.12
OS and version:    Mac OSX 10.9.5

## Actual behavior
VSCode with Python plugin is automatically inserting carriage return and breaking the words.  If I am tying def main():  example, the cursor is automatically advanced to the next line in the middle of the word as shown below:

de
f

this does not happen if there are lines with some words at the end.  This only happens as I am writing  a file that is new and does not have any existing line of either program code or comment at the end.  Also this does not happen in the Typescript code that I am writing for an Angular 2 project,  this is only happening with the Python code.  Note that this did not happen with the earlier version of the Python Plugin that I used to write another project code approx. 3 months ago.
## Steps to reproduce:  
-  
-  

## Settings  
Your launch.json (if dealing with debugger issues):  
```json
```   
Your settings.json:   
```
{
    "python.pythonPath": "/Users/Bharat/.pyenv/shims/python",
    "files.exclude": {
        "**/*.pyc": true
    },
    "vim.easymotion": true,
    "vim.incsearch": true,
    "vim.useSystemClipboard": true,
    "vim.hlsearch": true,
    "vim.useCtrlKeys": true,
    "vim.insertModeKeyBindings": [
        {
            "before": ["j", "j"],
            "after": ["<Esc>"]
        }
    ],
    "files.autoSave": "afterDelay",
    "files.trimTrailingWhitespace": true,
    "editor.renderWhitespace": "none",
    "editor.formatOnSave": true,
    "workbench.editor.enablePreview": false,
    "window.zoomLevel": 0,
    "editor.fontSize": 14
}
```   
## Logs
Output from ```Python``` output panel  
```
```   
Output from ```Console window``` (Help->Developer Tools menu)
```
```

**Comment by DonJayamanne** [comment]
Please could you confirm this issue doesn't occur when you disable the python plugin.

**Comment by bruparel** [comment]
Hello Don,
I am confirming that this issue does not occur when I disable the plugin.  And it started to re-occur as soon as enabled it again.  When I disabled it, I could still get the syntax hi-highlighting and indentation, but no intellisense or code completion.  Is there a way that I can install the previous version of the plugin and test it out?  Approx. 3 month's ago when I did write an approx. 1000 lines total Python program using your plugin, I did not have any issues.
Thanks.
Bharat

**Comment by bruparel** [comment]
Hello Don,
I think I may have found the problem.  In my settings.json file, I have the following line:

"editor.formatOnSave": true

I removed it and it started to work fine again.  I noticed it because I could see other strange behavior that was happening like the plugin insisting on putting three instead of two lines of white space between the imports and the start of a function definition of a file.  I heard your podcast with Michael Kennedy (very well done by the way) which tells me that you are calling other plugins under the hood.  I think one of the linters is not interacting properly with the setting above.  I may be wrong though.

Thanks for this great plugin.  Let me know if you want me to run any more tests at my end.
